dot/state/block.go

```diff
@@ -251,18 +253,12 @@ func (bs *BlockState) GetHeader(hash common.Hash) (*types.Header, error) {
 		return nil, err
 	}
 
-	rw := &bytes.Buffer{}
-	_, err = rw.Write(data)
-	if err != nil {
-		return nil, err
-	}
-
-	_, err = result.Decode(rw)
+	err = scale.Unmarshal(data, result)
 	if err != nil {
 		return nil, err
 	}
 
-	if reflect.DeepEqual(result, new(types.Header)) {
+	if reflect.DeepEqual(result, types.NewEmptyHeader()) {
```

Comment:
I was thinking about this and am not totally sure how to do that. Because types.NewEmptyHeader() initializes the digest vdt slice, I don't believe it == types.Header{}, and don't think it can be nil. And I don't believe VDT slices are comparable, so I don't think I can compare to VDT slices for equality. Do you have an ideas on where i'm wrong here/how else to approach it @timwu20 @noot?

Reply:
implement a custom method `(func h Header) Empty() bool` that checks that the number is 0 and ensures that the Digest (VDTSlice) is length 0.

Reply:
implemented. Added checks for the other fields as well because I ran into issues where it was all empty except for one or two of the hash fields